airbyte-integrations/connectors/source-mixpanel/source_mixpanel/manifest.yaml

```diff
@@ -203,13 +205,27 @@ definitions:
       name: cohort_members
       path: 2.0/engage
       field_path: results
+    state_migrations:
+      - type: CustomStateMigration
+        class_name: "source_mixpanel.components.CohortMembersLegacyToPerPartitionStateMigration"
     retriever:
       $ref: "#/definitions/retriever"
       requester:
         $ref: "#/definitions/requester"
         http_method: POST
       paginator:
         $ref: "#/definitions/paginator"
+      record_selector:
```

Comment:
I'm very confused by this. why can't we use a normal incrmental_sync component?

Reply:
It is not possible to filter records on API side.  So each time it returns all records and we filter them on our side.
 This behavior was migrated from the original python version of this connector.